server/src/main/java/com/cloud/network/vpc/VpcManagerImpl.java

```diff
@@ -3402,6 +3402,15 @@ public boolean isSrcNatIpRequired(long vpcOfferingId) {
                 && vpcOffSvcProvidersMap.get(Service.Gateway).contains(Network.Provider.VPCVirtualRouter));
     }
 
+    @Override
+    public boolean isSrcNatIpRequiredForVpcVr(long vpcOfferingId) {
+        final Map<Network.Service, Set<Network.Provider>> vpcOffSvcProvidersMap = getVpcOffSvcProvidersMap(vpcOfferingId);
+        return (Objects.nonNull(vpcOffSvcProvidersMap.get(Network.Service.SourceNat))
+                && vpcOffSvcProvidersMap.get(Network.Service.SourceNat).contains(Network.Provider.VPCVirtualRouter))
+                || (Objects.nonNull(vpcOffSvcProvidersMap.get(Network.Service.Gateway))
+                && vpcOffSvcProvidersMap.get(Service.Gateway).contains(Network.Provider.VPCVirtualRouter));
+    }
```

Comment:
how about `StaticNat` (and maybe others)?

Reply:
ok, might work. As you create a method `isSrcNatIpRequiredForVpcVr` the contents seems appropriate, but its use is more like `needsPublicInterface` does it? and in that case we would also want to test if the VR needs to provide `StaticNat`. I think a `Gateway` could be on a static ip as well, in theory at least.

maybe rename the method?

Reply:
for isolated networks (NATTED mode by default), it uses VR as SourceNat
for Routed networks, it uses VR as Gateway
in both cases, the VR needs a public IP (as Source NAT or Gateway)

For Static NAT, I think we do not need to add a public interface to the VR when VR is created.
when Static NAT is enabled, it may add a public interface if needed